src/erldns_resolver.erl

```diff
@@ -148,7 +148,7 @@ resolve_exact_type_match(Message, Qname, Qtype, Host, CnameChain, MatchedRecords
             false ->
               resolve_exact_type_match(Message, Qname, Qtype, Host, CnameChain, MatchedRecords, Zone, _AuthorityRecords, NSRecords)
           end;
-        {error, authority_not_found} ->
+        {error, _} ->
```

Comment:
I do not believe this should be chained to allow all error types to be handled this way, only the case where the authority is missing.